core/src/main/java/edu/ucr/cs/riple/core/Config.java

```diff
@@ -330,6 +344,11 @@ public Config(String[] args) {
     this.exhaustiveSearch = cmd.hasOption(exhaustiveSearchOption.getLongOpt());
     this.downStreamDependenciesAnalysisActivated =
         cmd.hasOption(downstreamDependenciesActivationOption.getLongOpt());
+    this.mode =
+        AnalysisMode.parseMode(
+            this.downStreamDependenciesAnalysisActivated,
+            cmd.getOptionValue(analysisMode),
+            !cmd.hasOption(analysisMode));
```

Comment:
Could simplify `parseMode` as described above and then here do:

```
    this.mode =
        AnalysisMode.parseMode(
            this.downStreamDependenciesAnalysisActivated,
            cmd.getOptionValue(analysisMode, "default"));
```

No need for the third argument or for supporting `null` modes.

Reply:
Simplified 56d8f5ed2ed954686f172272e79b425def7a30ee Thank you very much.